### PR TITLE
Protect circuit breaking functionality with an environment variable

### PR DESCRIPTION
**Pull request by donnadionne**
@donnadionne

---
This change is [Reviewable](https://reviewable.io/reviews/grpc/grpc/24396)

### CODE REVIEW

**Comment by donnadionne** [comment]
Protecting the circuit breaking (drop action) using environment variable.  All the logic of refcounting are left but will not result in any actions.

File: test/cpp/end2end/xds_end2end_test.cc [hunk 1]
```diff
@@ -2290,6 +2290,7 @@ TEST_P(XdsResolverOnlyTest, CircuitBreaking) {
     Status status_;
   };
 
+  gpr_setenv("GRPC_ENABLE_CIRCUIT_BREAKING", "true");
```
Comment (markdroth) [inline test/cpp/end2end/xds_end2end_test.cc:2293]: Please also add a test that the circuit breaking config is *not* honored when this env var is not set.

File: src/core/ext/filters/client_channel/lb_policy/xds/eds.cc [hunk 2]
```diff
@@ -56,6 +57,14 @@ constexpr char kEds[] = "eds_experimental";
 
 const char* kXdsLocalityNameAttributeKey = "xds_locality_name";
 
+bool XdsCircuitBreakingEnabled() {
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/lb_policy/xds/eds.cc:60]: Please add a TODO indicating that this should be removed as soon as we finish integration testing.

File: src/core/ext/filters/client_channel/lb_policy/xds/eds.cc [hunk 3]
```diff
@@ -56,6 +57,14 @@ constexpr char kEds[] = "eds_experimental";
...
+  char* value = gpr_getenv("GRPC_ENABLE_CIRCUIT_BREAKING");
```
[the remark this answers is not in the document]
Comment (voidzcy) [reply src/core/ext/filters/client_channel/lb_policy/xds/eds.cc:61]: By the way, I used `GRPC_XDS_EXPERIMENTAL_ENABLE_TIMEOUT` for timeout.

**Review by donnadionne — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/24396)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @GarrettGutierrez1 and @markdroth)

___
*[src/core/ext/filters/client_channel/lb_policy/xds/eds.cc, line 60 at r1](https://reviewable.io/reviews/grpc/grpc/24396#-MJURBAJkwxS8jmWY7r7-r1-60:-MJUappYDAMoBQ88gLYG:b-896fix) ([raw file](https://github.com/grpc/grpc/blob/a9a40bae5d1d1cf7b9fe2d12a451e559aa66650c/src/core/ext/filters/client_channel/lb_policy/xds/eds.cc#L60)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

Please add a TODO indicating that this should be removed as soon as we finish integration testing.
</blockquote></details>

Done.
___
*[src/core/ext/filters/client_channel/lb_policy/xds/eds.cc, line 61 at r1](https://reviewable.io/reviews/grpc/grpc/24396#-MJURBAJkwxS8jmWY7r7-r1-61:-MJUaqMS0XsuphL1Kium:b-896fix) ([raw file](https://github.com/grpc/grpc/blob/a9a40bae5d1d1cf7b9fe2d12a451e559aa66650c/src/core/ext/filters/client_channel/lb_policy/xds/eds.cc#L61)):*
<details><summary><i>Previously, voidzcy (Chengyuan Zhang) wrote…</i></summary><blockquote>

By the way, I used `GRPC_XDS_EXPERIMENTAL_ENABLE_TIMEOUT` for timeout.
</blockquote></details>

Done.
___
*[test/cpp/end2end/xds_end2end_test.cc, line 2293 at r1](https://reviewable.io/reviews/grpc/grpc/24396#-MJURBAJkwxS8jmWY7r8-r1-2293:-MJUhxjg8DVlBakNHwvo:b-896fix) ([raw file](https://github.com/grpc/grpc/blob/a9a40bae5d1d1cf7b9fe2d12a451e559aa66650c/test/cpp/end2end/xds_end2end_test.cc#L2293)):*
<details><summary><i>Previously, markdroth (Mark D. Roth) wrote…</i></summary><blockquote>

Please also add a test that the circuit breaking config is *not* honored when this env var is not set.
</blockquote></details>

Done.

**Review by markdroth — APPROVED** [review]
Looks good!  Feel free to merge after addressing the one remaining comment.

> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/24396#-:-MJXXXR1-PexkUt6hemO:b-vn51hs)* status: all files reviewed, 1 unresolved discussion (waiting on @donnadionne)

___
*[src/core/ext/filters/client_channel/lb_policy/xds/eds.cc, line 60 at r1](https://reviewable.io/reviews/grpc/grpc/24396#-MJURBAJkwxS8jmWY7r7-r1-60:-MJXXDr4DvppnNZ52jQ7:bk2iofs) ([raw file](https://github.com/grpc/grpc/blob/a9a40bae5d1d1cf7b9fe2d12a451e559aa66650c/src/core/ext/filters/client_channel/lb_policy/xds/eds.cc#L60)):*
<details><summary><i>Previously, donnadionne wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Please make the comment a TODO.

**Pull request merged**